### PR TITLE
[Enterprise Search] xpack/usage analytics collections count

### PR DESCRIPTION
**Pull request by sloanelybutsurely**
- updates enterprise search xpack/usage to include a count of analytics collections

### CODE REVIEW

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearchUsageTransportAction.java
Comment (ioanatia) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearchUsageTransportAction.java:94]: nitpick -should this be step 1? and should we also change the other comment `// Step 1: Fetch analytics collections count`.

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearchUsageTransportAction.java
Comment (ioanatia) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearchUsageTransportAction.java:103]: should we still do some error handling here?

Comment (sloanelybutsurely) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearchUsageTransportAction.java:103]: what do you mean? the error is handled on line 111.

if we throw any errors in this method we will break all of xpack/usage

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearchUsageTransportAction.java
Comment (sloanelybutsurely) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearchUsageTransportAction.java:94]: these are defined in reverse order so they can be chained. this is why i've added the comments so the actual execution order is clearer.

would you like me to change the order or execution?

Comment (ioanatia) [reply x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearchUsageTransportAction.java:94]: makes sense - the diff confused me a bit, found it easier to look at the entire file

**Review by ioanatia — APPROVED** [review]

**Review by joemcelroy — APPROVED** [review]
LGTM

**Review by jimczi — COMMENTED** [review]
The logic looks good but let's wait for #96104 to be merged. Sorry if it wasn't clear but we shouldn't merge directly in 8.8, I suggested to use the branch only to unblock testing locally.

**Comment by sloanelybutsurely** [comment]
> The logic looks good but let's wait for #96104 to be merged. Sorry if it wasn't clear but we shouldn't merge directly in 8.8, I suggested to use the branch only to unblock testing locally.

yeah that's fine. i can easily rebase against `main` once that's in. i just didn't want to have to wait for reviews

**Comment by sloanelybutsurely** [comment]
apologies for the force-push. had to switch base branches

**Comment by sloanelybutsurely** [comment]
@elasticmachine merge upstream

**Comment by sloanelybutsurely** [comment]
@elasticmachine merge upstream

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/application/EnterpriseSearchFeatureSetUsageSerializingTests.java
Comment (kderusso) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/application/EnterpriseSearchFeatureSetUsageSerializingTests.java:30]: Would it be worth moving some of the hard coded strings like `"count"` and moving them into variables we could reference?

File: x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/100_usage.yml
Comment (kderusso) [inline x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/100_usage.yml:1]: Nice tests 👍

**Review by kderusso — APPROVED** [review]
LGTM!

**Comment by sloanelybutsurely** [comment]
@elasticmachine merge upstream

**Comment by sloanelybutsurely** [comment]
@elasticmachine merge upstream

**Comment by sloanelybutsurely** [comment]
@elasticmachine test this please

**Comment by sloanelybutsurely** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.8|[](https://github.com/elastic/elasticsearch/pull/96495)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**
